### PR TITLE
test(qwen3.5): register Ornith-1.5-9B E2E manifest

### PR DESCRIPTION
**Pull request by roma5087**
## Background

Refs #1050. This PR does not close that issue — two of its acceptance
criteria remain unmet (see Exit Criteria).

`ornith-ai/Ornith-1.5-9B` is a public 9B reasoning checkpoint with
`model_type: qwen3_5` and architecture `Qwen3_5ForConditionalGeneration`. The
repository already has a `qwen3_5` family and native runtime path, but does not
list or validate this checkpoint.

This PR registers the checkpoint in the E2E test inventory at a pinned,
immutable revision. It adds no runtime, builder, or family code.

## Exit Criteria

- [x] The exact checkpoint and an immutable revision are registered in the test
      inventory (`hf_revision` pinned to `489cb97981b8654bcfcf30ce1f94ed1b62e07b53`).
- [x] An FP16 engine builds from the BF16 source through the public builder
      with no manual checkpoint rewriting.
- [x] The bundle runs through the native runtime on an H100 80GB for a
      representative chat prompt.
- [x] No test threshold, passing criterion, or validator is weakened. The
      threshold sidecar overrides nothing (`"threshold_overrides": {}`), so the
      case runs on family defaults.
- [x] The checkpoint appears in generated supported-model documentation via its
      `website/data/hf-model-metadata.json` entry.
- [ ] A tool-calling prompt is validated against upstream Transformers. **Not
      met** — see Not Run / Remaining Gaps.
- [x] The registered E2E passes. This required the family fix in `4fadf73f`
      ("fix(qwen3_5): serialize decoder config"), which landed after this PR was
      opened; see Not Run / Remaining Gaps for what I verified and when.

Non-goals: image/video input (the checkpoint carries a Qwen3.5 vision config;
this PR covers the text path only), long-context validation beyond
`max_cache_length: 256`, and any benchmark/accuracy workload binding.

## Implementation

Eight files, no product code:

- `tests/e2e/models/qwen3_5/manifests/ornith-1-5-9b.json` (new) — modelled on
  the sibling `qwen35-9b.json`, differing only in `name`, `hf_id`, `bundle`,
  `trace_id`, `notes`, and the addition of `hf_revision`.
- `tests/e2e/models/qwen3_5/MODEL.toml` — one line registering the manifest.
- `website/data/hf-model-metadata.json` — one checkpoint entry, inserted in
  sorted position, `revision` matching the manifest's `hf_revision` as
  `website/plugins/model-support-inventory/index.js:1424` requires.
- `tests/e2e/models/qwen3_5/thresholds/ornith-1-5-9b.json` (new) — required
  by `test_model_owned_e2e_assets_are_local_and_complete` and
  `test_repo_model_thresholds_are_sidecars` for any case that is not
  invariant-only. It sets `"threshold_overrides": {}` so no tolerance is
  changed; the case passes on family defaults.
- `tests/validation/model_workloads.yaml` — one binding
  (`workloads: [mmlu_five_shot_mcq]`, matching the sibling `qwen35-9b`),
  required by `test_model_workload_catalog_covers_every_ready_model`. The
  binding declares which workload applies; I have not executed it.
- `benchmarks/performance/release.yaml` — one `excluded_profiles` entry.
  Four tests (`test_release_suite_covers_every_non_l0_ready_model_profile`,
  `test_ready_models_have_release_performance_coverage_or_exclusion`,
  `test_all_accuracy_selects_only_accuracy_catalog_models`,
  `test_suite_timing_contract_drift_is_rejected_before_execution`) all fail
  with the same message, `missing=ornith-1-5-9b`, until a ready model has
  either release-performance coverage or a declared exclusion. I have no
  release-performance receipt to contribute, so the exclusion is the honest
  option; the wording follows the existing LFM2 entries.
- `tests/tools/test_perf_matrix.py` — the expected-exclusion inventory that
  mirrors `release.yaml`: an `ORNITH_EXCLUSION_REASON` constant, an entry in
  the `excluded_profiles` assertion, an entry in `explicit_exclusions`, and
  the exclusion count 6 -> 7. The reason string is byte-identical to the
  YAML, which the assertion compares directly.
- `tests/tools/test_trtmc_validate.py` — three hardcoded inventory counts
  bumped to include this model (119 -> 120, 120 -> 121, 119 -> 120 after the
  rebase onto current main). This is the established practice for adding a model;
  the same line went 105 -> 106 -> 108 -> 109 -> 114 -> 115 in the
  minimax-h3, dinov3, stereo, lfm2 and voicechat additions. No assertion
  was relaxed — the counts are an inventory the test exists to enforce.

On acceptance criterion 5 (whether `qwen3_5` is reused unchanged, or which
differences require support):

- **Config**: 53 fields identical to `Qwen/Qwen3.5-9B`; 3 differ in value
  (`text_config.use_cache`, `transformers_version`, `vision_config.model_type`);
  9 exist only in Ornith. None are read differently by the family.
- **Tokenizer**: `tokenizer.json`, `vocab.json`, `merges.txt` and
  `tokenizer_config.json` are byte-identical to `Qwen/Qwen3.5-9B` by sha256.
- **Chat template**: `chat_template.jinja` differs in exactly one branch, which
  renders prior assistant turns. Single-turn prompts render identically.
- **Weights**: identical tensor names and shapes. One real difference: for each
  of the 24 DeltaNet layers, `linear_attn.A_log` and `linear_attn.norm.weight`
  are BF16 in Ornith where `Qwen/Qwen3.5-9B` stores them FP32 (48 tensors,
  3,840 elements). Flagged because #725 preserved FP32 DeltaNet state
  deliberately; I have not measured any accuracy impact.

No checkpoint-name-only behavior was added.

### Change categories

- [x] Model or runtime behavior
- [ ] Public API
- [ ] ABI
- [ ] Bundle or artifact format
- [ ] Dependencies
- [ ] Documentation only
- [ ] CI or developer tooling

### Commands and Results

Build:

```
trtmc build ornith-ai/Ornith-1.5-9B --precision fp16 --max-cache-length 256 \
  -o ornith-1-5-9b.bundle

[trtmc build] Model: qwen3_5 (layers=32, hidden=4096, vocab=248320)
[trtmc build] Family: qwen3_5
[trtmc build] Weights loaded [245.2s]
[trtmc build] Engine built [583.2s] (17081.7 MB)
[trtmc build] Bundle saved [861.2s total]

trtmc inspect ornith-1-5-9b.bundle --list-engines
  engine_plan   17081.7 MB   primary
```

Run:

```
trtmc run ornith-1-5-9b.bundle \
  --prompt "What is the capital of France? Answer in one word." \
  --chat-template --max-new-tokens 200

[trtmc] Pipeline loaded (strategy=qwen3_5_hybrid_mamba_attention, backend=trt_new_runtime)
[trtmc-perf] Prefill: 20 tokens, 148.5 ms (134.7 tok/s)
[trtmc-perf] Decode:  67 steps, 490.9 ms (136.5 tok/s, 7.33 ms/tok)

<think> ... </think>

Paris
```

Registered E2E:

```
python3 -m pytest tests/e2e -k ornith --engine-dir <dir> \
  --trtmc-binary build-sm90/trtmc --hf-python /usr/bin/python3 \
  --model-plugin-dir build-sm90/models -q
```

- On unmodified `main`: **FAILS**.
  `compare_fail … NED=0.875`, TRT `"paris is the capital of france. it's the"`
  vs reference `"paris"`.
- With the family defect described below corrected: **PASSES**.
  `result.json` → `status: pass`; TRT emits `[57590, 248046]`
  ("Paris", `<|im_end|>`), a token-for-token match against the fp32 reference.

Static validators (this branch, this change set):

```
PYTHONPATH=python:. python3 tools/model_ci.py validate      -> ok
PYTHONPATH=python:. python3 tools/test_impact.py --validate -> Validation passed
git diff --check                                            -> clean
```

### Hardware, Environment, and Revisions

- Validation was performed at `7aa0b211e52ad17f6d827d4c33022b36130e7f01`. The
  branch has since been rebased onto `af40d433ea1b` (current main). No commit
  in that range touches `src/runtime/models/qwen3_5/`,
  `tests/e2e/models/qwen3_5/`, `engine_builder.py`, or `json_helpers.cpp`, so
  the recorded results still describe this code path; the rebase only
  refreshed inventory counts that other model additions had moved.
- Checkpoint: `ornith-ai/Ornith-1.5-9B` @ `489cb97981b8654bcfcf30ce1f94ed1b62e07b53`
- Reference: `Qwen/Qwen3.5-9B` @ `c202236235762e1c871ad0ccb60c8ee5ba337b9a`
- GPU: NVIDIA H100 80GB HBM3, compute capability 9.0, driver 580.126.09
- TensorRT: 11.1.0.106, from both apt and pip, pinned to match. The apt
  packages are the `11.1.0.106-1+cuda13.3` builds; `libnvinfer.so` has no
  dynamic CUDA dependencies, so that tag reflects what TensorRT was built
  against rather than a runtime library it loads.
- CUDA toolkit (nvcc): 12.8.93. No CUDA 13 toolkit was installed; the driver
  (580.126.09) supports CUDA 13.
- transformers 5.16.1; Python 3.12
- Build: host build (no container),
  `-DCMAKE_BUILD_TYPE=Release -DCMAKE_CUDA_ARCHITECTURES=90-real`
- Engine precision fp16; reference precision fp32; `max_cache_length` 256

### Not Run / Remaining Gaps

**The E2E result recorded above was obtained before `4fadf73f` landed.** When
this PR was opened, the registered E2E failed on `main`: the bundle's
`config.json` was written from the unmerged HF config, while the C++ runtime
reads config keys at the top level only (`src/utils/json_helpers.cpp:38`), so
checkpoints nesting decoder dims under `text_config` resolved to struct
defaults. `Qwen/Qwen3.5-9B` got `kv_dim = 0` and failed with `Failed to create
Qwen35KvCache for hybrid model`; Ornith got `kv_dim = 4096` against an engine
binding 1024, producing a correct first token and divergence thereafter. I
verified on an H100 that lifting the nested scalars made both run, and made
this manifest pass with an exact token match (`[57590, 248046]`) against the
fp32 reference.

`4fadf73f` fixes this upstream by materializing those fields in the family
plugin's bundle-config overrides. I have rebased onto it but have **not**
re-run the GPU E2E against the rebased tree — I no longer have the H100. The
internal CI run is the authoritative check.

**Tool calling is not validated.** Acceptance criterion 4 asks for a
tool-calling prompt. The public CLI has no tool-specification flag and the C++
ChatML path does no tool-call serialization, so this cannot be exercised
through the public runtime interface today. Recording it as a boundary, per the
issue's "within the capabilities of the public runtime interface".

**`tests/e2e/test_full_pipeline.py::test_full_pipeline[ornith-1-5-9b]` fails**
at `diff_logits` with `max_abs_logit_diff 23.4` and `!!!!!!!!!!!` output on a
multi-turn prompt. This appears to be in the Python runner that `diff_logits`
compares against HF, not the bundle: the identical prompt through the C++
binary returns `" 2+2 equals 4."`, matching the reference. I could not
determine whether `qwen35-9b` fails the same way, because it cannot run at all
on `main` (above). Unresolved.

**Protected CI not run.** I cannot apply `run-internal-ci`, so the GPU E2E has
only been run on my own H100.

**Multimodal and long context untested.** Text path only;
`max_cache_length: 256`, far below the declared 262,144 context.

**Ornith's `kv_dim` was measured only by derivation.** I printed the parsed
config for `Qwen/Qwen3.5-9B` directly; for Ornith the value follows from the
same parser and its config, but I did not print it.

## Contributor Self-Review

- [x] I have completed a self-review of this change.

## Notes For Future Readers

- The manifest pins `hf_revision`, which the sibling `qwen35-9b.json` does not.
  `orchestrator.py:355` forwards it as `--model-revision`.
- The threshold sidecar is intentionally empty. Mirroring `qwen35-9b.json`
  would loosen `logit_atol` from 0.001 to 0.005; the case passes on defaults,
  so the looser tolerance is unnecessary. If you would rather it mirror the
  sibling exactly, say so and I will change it.
- `trust_remote_code: true` matches the sibling manifest and is what the
  validated run used. The checkpoint ships no `.py` files, so it is a no-op; I
  kept it rather than ship a value I had not tested.
- Licence: the model card declares `license: mit`, but its `license_link` 404s
  and no LICENSE file is published in the repo. The vendor blog states no
  licence. Flagging rather than asserting.
- The 48 BF16-vs-FP32 DeltaNet tensors noted above are checkable in one call:
  the HF API `?expand[]=safetensors` reports `F32: 3840` for `Qwen/Qwen3.5-9B`
  and no F32 for Ornith.
- Lifting `eos_token_id` (in the fix described above) gives `Qwen/Qwen3.5-9B`
  `248044` (`<|endoftext|>`) rather than the tokenizer's `<|im_end|>`
  (`248046`). Generation stopped correctly in my test, but I am not confident
  that is the right value for a chat model.

### Risk level

- [x] Low
- [ ] Medium
- [ ] High

Test-inventory registration plus one documentation-metadata entry. No runtime,
builder, or family code changes. The only behavioral effect is that one
additional E2E case is collected.

### CODE REVIEW

**Comment by yifeif-nv** [comment]
Hi @roma5087 Looks like this model is just sharing the same Qwen3.5 architecture! The PR looks good to me. I'll trigger the internal CI for it

**Comment by yifeif-nv** [comment]
Hi @roma5087 just to follow up on this PR. Can you rebase the PR onto the latest main TOT so we can retrigger the internal CI and push it through to get it merged

File: apps/benchmark/performance/release.yaml [hunk 1]
```diff
@@ -51,7 +56,7 @@ excluded_profiles:
   - model: res2net50d-in1k
     reason: &res2net_performance_exclusion >-
       Functional and timm reference-parity qualification is present for every
-      published Res2Net checkpoint, but the release-performance workload and
+      published Res2Net checkpoint, but the release-pwherformance workload and
```
[the remark this answers is not in the document]
Comment (yifeif-nv) [reply apps/benchmark/performance/release.yaml]: Real typo. Please fix this and I'll trigger the internal CI for your!
